config/routes.rb

```diff
@@ -1,5 +1,14 @@
+# require 'devise_token_auth'
+
 Rails.application.routes.draw do
 
   root to: 'main#index'
+  namespace :api do
+    scope :v1 do
+      mount_devise_token_auth_for 'User', at: 'auth'
+    end
+  end
+
+  get '/api/service/get-book-isbn', to: 'open_library#book'
```

Comment:
move this inside namespace `api/v1` and change
```suggestion
resources :open_library, only: :show 
```

https://guides.rubyonrails.org/routing.html